src/main/java/twilightforest/block/BlockTFNagastone.java

```diff
@@ -82,403 +55,317 @@ public int getRenderType() {
+    // metadata = 4;
+    // }
+    // if (metadata > 0 && (direction == 0 || direction == 1) && par1World.getBlock(x, y + 1, z) == this) {
+    // metadata = 8;
+    // }
+    // // if we have decided not to place a straight block, face the block in question towards the player
+    // if (metadata == 0 || metadata == 4 || metadata == 8)
+    // {
+    // direction = BlockTFNagastone.determineOrientation(par1World, x, y, z, (EntityPlayer)par5EntityLiving);
+    // }
+    //
+    // // if we are placing a block up and down, see if there is block to the side, and if so, modify to connect
+    // if (metadata > 0 && direction == 2)
+    // {
+    // int connect = getOnlyNSEWConnection(par1World, x, y, z);
+    //
+    // if (connect != -1)
+    // {
+    // // recall that pistonOrient variable to determine if we're connecting up or down
+    // metadata = pistonOrient == 0 ? 4 : 8;
+    // direction = connect;
+    // }
+    // }
+    //
+    // // if, after all this, we've decided to place a straight block, and there is only one place to connect it,
+    // connect it
+    // if (metadata == 12)
+    // {
+    // int connect = getOnlyConnection(par1World, x, y, z);
+    // //System.out.println("Trying to orient straight block. Connection = " + connect);
+    //
+    //
+    // if (connect != -1)
+    // {
+    // switch (connect)
+    // {
+    // case 0:
+    // case 1:
+    // direction = 1;
+    // break;
+    // case 2:
+    // case 3:
+    // direction = 0;
+    // break;
+    // case 4:
+    // case 5:
+    // direction = 2;
+    // }
+    // }
+    // }
+    //
+    // // I think we've covered it
+    // par1World.setBlockMetadataWithNotify(x, y, z, metadata | direction, 3);
+    // }
 
+    /**
+     * Called when a block is placed using its ItemBlock. Args: World, X, Y, Z, side, hitX, hitY, hitZ, block metadata
+     * 
+     * 0 - 3 head directions 4 - 7 bend down 8 - 11 bend up 12 straight n/s 13 straight e/w 14 straight u/d 15 weird
+     * middle piece
+     */
     @Override
-    public TileEntity createTileEntity(World world, int metadata) {
-        return new TileEntityTFNagastone();
+    public int onBlockPlaced(World world, int x, int y, int z, int side, float hitX, float hitY, float hitZ,
+            int metadata) {
+        int type = metadata & 12;
+        int orient = 0;
+        if (type == 0) {
+            // head!
+            orient = switch (side) {
+                default ->
+                    // don't know
+                    1;
+                case 2 -> 0;
+                case 3 -> 1;
+                case 4 -> 2;
+                case 5 -> 3;
+            };
+        }
+        if (type == 12) {
+            // straight
+            // n/s
+            orient = switch (side) {
+                case 0, 1 -> 2; // u/d
+                case 2, 3 -> 1; // e/w
+                case 4, 5 -> 0;
+                default -> orient;
```

Comment:
Please do not use magic numbers for directions. Use the ForgeDirection enum class. It also has helpers for rotations so no need to reinvent the wheel.

Reply:
That's an original Benimatic's code, lol